src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -136033,6 +136034,236 @@ partial class C2<T> : Base<
                 );
         }
 
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/40538")]
+        public void AnnotationMismatch_Interface_GenericClass_01()
+        {
+            var source = """
+                #nullable enable
+                var c = new C<int>();
+
+                partial class C<T> : I<T> { }
+
+                partial class C<T> :
+                    #nullable disable
+                    I<T>
+                    #nullable enable
+                {
+                }
+
+                public interface I<T> { }
+                """;
+
+            var verifier = CompileAndVerify(CreateCompilationWithMscorlib40(source)).VerifyDiagnostics();
+
+            verifier.VerifyIL("<top-level-statements-entry-point>", """
+                {
+                  // Code size        7 (0x7)
+                  .maxstack  1
+                  IL_0000:  newobj     "C<int>..ctor()"
+                  IL_0005:  pop
+                  IL_0006:  ret
+                }
+                """);
+
+            verifier.VerifyTypeIL("C`1", """
```

Comment:
Here the primary goal was to capture the existing IL codegen and that nothing has changed with it. I can add more targeted symbol validation.

Reply:
> Here the primary goal was to capture the existing IL codegen and that nothing has changed with it. I can add more targeted symbol validation.

Given the nature of the change in this PR, it would be impossible to affect metadata or IL with it. Therefore, I suggest getting rid of all IL validations as unnecessary and confusing for the reader/maintainer of the test.